### PR TITLE
perf: api request optimisations

### PR DESCRIPTION
**Pull request by 0xApotheosis**
## Description

- No longer fetches quotes when the sell amount is 0, and updates UI to hide related elements when this is the case
- Updates polling for the `tradeQuoteQuery` to only poll if the window is currently in focus

Note to devs, we won't poll quotes now if you're focused on dev console.

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [x] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

Closes https://github.com/shapeshift/web/issues/4989

## Risk

- Risk that we don't get quotes whne we should
- Risk that we show/don't show UI when we should/shouldn't

## Testing

Check swapper for all combinations of:
- when a 0 sell amount is entered, and not entered
- when window is focused, and not focused

### Engineering

☝️

### Operations

☝️

## Screenshots (if applicable)

N/A

### CODE REVIEW

**Comment by 0xApotheosis** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #4992** <a href="https://app.graphite.dev/github/pr/shapeshift/web/4992" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/4992?utm_source=stack-comment).

File: src/components/MultiHopTrade/hooks/quoteValidation/useActiveQuoteStatus.tsx [hunk 1]
```diff
@@ -74,7 +83,7 @@ export const useActiveQuoteStatus = (): QuoteStatus => {
         case ActiveQuoteStatus.NoConnectedWallet:
           return 'common.connectWallet'
         case ActiveQuoteStatus.BuyAssetNotNotSupportedByWallet:
-          return ['trade.quote.noReceiveAddress', { assetSymbol: tradeBuyAsset?.symbol }]
+          return ['trade.errors.noReceiveAddress', { assetSymbol: tradeBuyAsset?.symbol }]
```
Comment (0xApotheosis) [inline src/components/MultiHopTrade/hooks/quoteValidation/useActiveQuoteStatus.tsx:86]: An unrelated fix, but something I noticed whilst testing.

File: src/components/MultiHopTrade/hooks/useGetTradeQuotes.tsx [hunk 2]
```diff
@@ -163,8 +167,16 @@ export const useGetTradeQuotes = () => {
     receiveAccountMetadata?.bip44Params,
   ])
 
+  useEffect(() => {
+    const interval = setInterval(() => {
+      console.log('document.hasFocus()', document.hasFocus())
```
Comment (0xdef1cafe) [inline src/components/MultiHopTrade/hooks/useGetTradeQuotes.tsx:172]: delet

Comment (0xApotheosis) [reply src/components/MultiHopTrade/hooks/useGetTradeQuotes.tsx:172]: derp

Comment (0xApotheosis) [reply src/components/MultiHopTrade/hooks/useGetTradeQuotes.tsx]: https://github.com/shapeshift/web/pull/4992/commits/9862be62eb6bd02dae4a5962a15551f25cfa5fbd

**Review by 0xdef1cafe — APPROVED** [review]
does what it says elegantly and correctly

**Pull request merged**